src/Foundatio.Storage.SshNet/Storage/SshNetFileStorage.cs

```diff
@@ -131,17 +128,26 @@ public SshNetFileStorage(Builder<SshNetFileStorageOptionsBuilder, SshNetFileStor
         _logger.LogInformation("Renaming {Path} to {NewPath}", normalizedPath, normalizedNewPath);
         EnsureClientConnected();
 
+        if (await ExistsAsync(normalizedNewPath).AnyContext()) {
+            _logger.LogDebug("Removing existing {NewPath} path for rename operation", normalizedNewPath);
+            await DeleteFileAsync(normalizedNewPath, cancellationToken).AnyContext();
+            _logger.LogDebug("Removed existing {NewPath} path for rename operation", normalizedNewPath);
+        }
```

Comment:
This is now required because there is a really weird failure they throw if the new path already exists. We were using an overload that had it do a posix rename which I'm guessing just did an overwrite of the file.